Codes of Real-Time IoT Perception Systems Based on Edge-Cloud Collaboration with Large Foundation Models

hsj576 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://github.com/kubeedge/ianvs/blob/main/CONTRIBUTING.md
2. Ensure you have added or ran the appropriate tests for your PR

-->

**What type of PR is this?**
/kind feature
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind test
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind failing-test
-->


**What this PR does / why we need it**:
This PR provides codes of Real-Time IoT Perception Systems Based on Edge-Cloud Collaboration with Large Foundation Models.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #80

bf-yang commented:
/lgtm

MooreZheng commented:
After the presentation in the routine meeting, the overall result looks fine to us. 

The project would become even better if 
1. Might combine the OSPP project with the LFX one, since they are using the same dataset, with the same scheme, and under the same task. 
2. Add comments, APIs, and tutorials for the project. 
3. Add docs of scenarios, algorithms, leaderboards, and test reports on https://ianvs.readthedocs.io/en/latest/

hsj576 commented:
> After the presentation in the routine meeting, the overall result looks fine to us.
> 
> The project would become even better if
> 
> 1. Might combine the OSPP project with the LFX one, since they are using the same dataset, with the same scheme, and under the same task.
> 2. Add comments, APIs, and tutorials for the project.
> 3. Add docs of scenarios, algorithms, leaderboards, and test reports on https://ianvs.readthedocs.io/en/latest/

I made the following modifications: 
1. Merged the code for the OSPP project and the LFX project, now they are both located under examples/robot. 
2. Added test reports related to the Cloud-Robotics dataset under docs/leaderboards. 
3. Added comments for the APIs, code interfaces in the form of python docstrings.

jaypume commented on examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py:
```diff
@@ -0,0 +1,209 @@
+            print("predict start for small data")
+            if not isinstance(data[0][0], dict):
+                data = self._preprocess(data)
+                #print("predict starting 69")
```
The lines for local debug should not be committed, these TEMP PRINT lines should be deleted.

hsj576 replied:
TEMP PRINT lines are removed.

jaypume commented on examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py:
```diff
@@ -0,0 +1,209 @@
+            self.validator.test_loader = DataLoader(data, batch_size=self.val_args.test_batch_size, shuffle=False,
+                                                pin_memory=True)
+        
+        #print("predict starting 75")
```
All these TEMP PRINT lines should be deleted

hsj576 replied:
TEMP PRINT lines are deleted.

hsj576 commented on examples/robot/lifelong_learning_bench/testalgorithms/rfnet/RFNet/utils/args.py:
```diff
@@ -21,7 +21,7 @@ def __init__(self, **kwargs):
 
         self.seed = 1
-        self.resume = None
+        self.resume = '/home/hsj/ianvs/project/pretrain/pretrain_model.pth'
```
The hardcode path is removed.

hsj576 commented on examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py:
```diff
@@ -0,0 +1,209 @@
+        for epoch in range(epoch_num):
+            train_loss = self.trainer.my_training(epoch)
+            loss_all.append(train_loss)
+        with open('/home/shijing.hu/ianvs/project/ianvs/train_loss_2.txt', 'a+') as file:
```
The hardcode path is removed.

jaypume commented:
Please squash these commits to make commit history more clean, and the better count should be 3-5 commits.
You can just do it by squashing the duplicate commits.

hsj576 commented:
> Please squash these commits to make commit history more clean, and the better count should be 3-5 commits. You can just do it by squashing the duplicate commits.

The duplicate commits are squashed.

jaypume commented:
/lgtm

jaypume commented:
/approve
